internal/v2/s2av2.go

```diff
@@ -69,14 +71,24 @@ func (c *s2av2TransportCreds) ClientHandshake(ctx context.Context, serverAuthori
 	if err != nil {
 		return nil, nil, err
 	}
+	// Create an AccessTokenManager instance to use to authenticate to S2Av2.
+	os.Setenv("S2A_ACCESS_TOKEN", "token")
+	accessTokenManager, err := tokenmanager.NewSingleTokenAccessTokenManager()
+	if err != nil {
+		return nil, nil, err
+	}
+	token, err := accessTokenManager.DefaultToken()
+	if err != nil {
+		return nil, nil, err
+	}
 	var config *tls.Config
 	if c.serverName == "" {
-		config, err = tlsconfigstore.GetTlsConfigurationForClient(serverName, cstream)
+		config, err = tlsconfigstore.GetTlsConfigurationForClient(serverName, cstream, token)
```

Comment:
I think the model should be: pass the token manager pointer to whatever API needs the token (so GetTlsConfigurationForClient, GetTlsConfigurationForServer), if that API has a local identity then get the token for that local identity and if not use the default, and then attach the token to the request.